### PR TITLE
Enhancement: polygon2d right click to undo last point, delete to cancel

### LINKED ISSUE #583: Allow to undo last point when creating polygon 2D
**Describe the project you are working on:**
Game with lots of complex polygons.

**Describe the problem or limitation you are having in your project:**
So, polygons are main nodes building the environment collisions in my game. There's lots of them and they have lots of vertices (like, tens of them). I have my maps drawn based on grid, so when I create polygon I enable grid and join points on the texture. BUT, this requires lots of precision. And I sometimes make a mistake (i.e. place a point off by a grid cell). Theoretically not a big deal, because I can just correct it after finishing the polygon. But sometimes I make few mistakes and while correcting I might omit one and this results in a disaster (not really, but the collider doesn't match and I usually find it when testing game, but imagine I totally overlooked it. THAT would be disaster).

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Would be nice if I could undo last point when drawing polygon. Right now, right-click will cancel whole polygon. This is rarely really useful and also could be done with Escape. So right-click could remove last point instead, so when I make a mistake I can quickly correct it.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Just remove last polygon point when right-clicking during drawing.
Uno

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
It can't.

**Is there a reason why this should be core and not an add-on in the asset library?:**
It extends existing functionality.

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: https://github.com/godotengine/godot-proposals/issues/583

### CODE REVIEW

File: editor/plugins/abstract_polygon_2d_editor.cpp [hunk 1]
```diff
@@ -466,7 +466,26 @@ bool AbstractPolygon2DEditor::forward_gui_input(const Ref<InputEvent> &p_event)
...
+				} else {
+
+					const Vertex active_point = get_active_point();
+
+					if (active_point.valid()) {
+
+						remove_point(active_point);
+						return true;
+					}
+				}
```
Comment (KoBeWi) [inline editor/plugins/abstract_polygon_2d_editor.cpp:488]: This duplicates Backspace action code, should be moved to a method.

File: editor/plugins/abstract_polygon_2d_editor.cpp [hunk 2]
```diff
@@ -555,6 +574,8 @@ bool AbstractPolygon2DEditor::forward_gui_input(const Ref<InputEvent> &p_event)
 					return true;
 				}
 			}
+		} else if (k->get_keycode() == KEY_DELETE) {
+			_wip_cancel();
```
Comment (KoBeWi) [inline editor/plugins/abstract_polygon_2d_editor.cpp:578]: I think delete can be outright removed. It doesn't work correctly (deletes the node) and you can just use Escape.

**Comment by KoBeWi** [comment]
Needs rebase.

Here's a patch with resolved conflicts (I needed it for testing): 
[rebased.txt](https://github.com/godotengine/godot/files/9592862/rebased.txt)

**Comment by Calinou** [comment]
I tested @KoBeWi's version locally. Right-click works as expected but <kbd>Delete</kbd> brings up the node deletion dialog.

**Pull request closed without merging**